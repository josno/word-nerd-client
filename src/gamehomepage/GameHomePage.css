.game-container {
	margin-top: 5rem;
}

.list-of-games {
	border: 2px solid black;
	font-size: 18px;
	text-align: left;
	padding-left: 0.4rem;
	margin-left: auto;
	margin-right: auto;
	width: 70vw;
}

.create-game-button {
	margin: 4rem;
	color: brown;
}
