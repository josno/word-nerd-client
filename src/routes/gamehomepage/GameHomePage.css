.game-container {
	margin-top: 5rem;
	display: flex;
	flex-direction: column;
}

.create-game-button {
	margin: 4rem;
	color: brown;
}

.create-game-button:hover {
	color: blue;
	cursor: pointer;
}
