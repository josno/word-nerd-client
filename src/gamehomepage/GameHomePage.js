import React, { Component } from 'react';
import './GameHomePage.css';
import { Link } from 'react-router-dom';
import WordContext from '../WordContext';

class GameHomePage extends Component {
	static contextType = WordContext;

	render(props) {
		const { savedGames, getSavedGameId } = this.context;
		return (
			<div>
				<section>
					<div className="game-container">
						<h1> Games List </h1>
						<ul className="list-of-games">
							{savedGames.length > 0 &&
								savedGames.map(g => (
									<li
										key={`${g.id}-${g.title}`}
										className="game-info"
									>
										<div className="info-container">
											<h2>{g.title}</h2>
											<h3>
												{g.date_created.toLocaleDateString()}
											</h3>
										</div>

										<div className="button-container">
											<Link
												to={`/game-start-page/${g.game_id}`}
											>
												<button
													className="play-button"
													type="submit"
													onClick={() => {
														getSavedGameId(
															g.game_id
														);
													}}
												>
													Play
												</button>
											</Link>
										</div>
									</li>
								))}
						</ul>

						<Link to="/input-page">
							<button className="create-game-button">
								<h2>Create a Pass the Ball Game</h2>
							</button>
						</Link>
					</div>
				</section>
			</div>
		);
	}
}

export default GameHomePage;
