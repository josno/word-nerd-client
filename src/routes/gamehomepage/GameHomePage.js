import React, { Component } from 'react';
import './GameHomePage.css';
import { Link } from 'react-router-dom';
import WordContext from '../../WordContext';
import config from '../../config';
import GamesList from '../../components/gameslist/GamesList';
import TokenService from '../../services/token-service';
// import GamesService from '../services/api-service';

class GameHomePage extends Component {
	static contextType = WordContext;
	constructor(props) {
		super(props);
		this.state = {
			savedGames: [],
			noGamesSaved: false,
			user_id: ''
		};
	}

	componentDidMount() {
		fetch(`${config.API_ENDPOINT}/v1/games/`, {
			headers: {
				Authorization: `bearer ${TokenService.getAuthToken()} `
			}
		})
			.then(res => res.json())
			.then(responsejson => {
				if (responsejson.length === 0) {
					this.setState({
						noGamesSaved: true
					});
				} else {
					this.setState({
						savedGames: [...responsejson],
						user_id: responsejson[0]['user_id']
					});
				}
			});
	}

	render(props) {
		const { savedGames, user_id } = this.state;

		return (
			<div>
				<section>
					<div className="game-container">
						<h1> Games List </h1>

						{this.state.noGamesSaved === true ? (
							<div className="info-container">
								You have no saved games yet!
							</div>
						) : (
							<GamesList savedGames={savedGames} />
						)}
						<Link to={`/input-page/${user_id}`}>
							<button className="create-game-button">
								<h2>Create a Pass the Ball Game</h2>
							</button>
						</Link>
					</div>
				</section>
			</div>
		);
	}
}

export default GameHomePage;
