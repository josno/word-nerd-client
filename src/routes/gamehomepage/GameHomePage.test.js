import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import GameHomePage from './GameHomePage';
import renderer from 'react-test-renderer';

describe('<Login/>', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<Router>
				<GameHomePage />
			</Router>,
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	});
	it('renders as expected', () => {
		const tree = renderer
			.create(
				<Router>
					<GameHomePage />
				</Router>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
