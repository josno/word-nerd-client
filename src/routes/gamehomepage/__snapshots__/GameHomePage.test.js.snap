// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> renders as expected 1`] = `
<div>
  <section>
    <div
      className="game-container"
    >
      <h1
        className="games-list-title"
      >
         Games List 
      </h1>
      <div
        className="game-list-container"
      >
        <ul
          className="list-of-games"
        />
      </div>
      <button
        className="create-game-button"
      >
        <a
          href="/input-page"
          onClick={[Function]}
        >
          <h2>
            Create a Pass the Ball Game
          </h2>
        </a>
      </button>
    </div>
  </section>
</div>
`;
